pymeasure/instruments/hp/hp3478A.py

```diff
@@ -37,126 +37,68 @@
+        # ("ERR_ROM", c_uint8, 1),
+        # ("ERR_RAM", c_uint8, 1),
+        # ("ERR_cal", c_uint8, 1),
+        ("Error_Status", c_uint8, 8),
         # Byte 5: DAC Value
-        ("DAC_value",       c_uint8, 8),
+        ("DAC_value", c_uint8, 8),
     ]
 
-    def __str__(self):
-        """
-        Returns a pretty formatted (human readable) string showing the status of the instrument
-
-        """
-        cur_mode = HP3478A.INV_MODES["F" + str(self.function)]
-        cur_range = list(HP3478A.RANGES[cur_mode].keys())[self.range - 1]
-        if cur_range >= 1E6:
-            r_str = str(cur_range / 1E6) + ' M'
-        elif cur_range >= 1000:
-            r_str = str(cur_range / 1000) + ' k'
-        elif cur_range <= 1:
-            r_str = str(cur_range * 1000) + ' m'
-        else:
-            r_str = str(cur_range) + ' '
-        return (
-            "function: {}, range: {}, digits: {}\
-                \nStatus:\n  internal | external trigger: {} | {}\
-                \n  Auto ranging: {}\n  AutoZero: {}\
-                \n  50Hz mode: {}\n  Front/Rear selection: {} \
-                \n  Calibration enable: {}\
-                \nSerial poll mask (SRQ):\n  SRQ for Data ready: {}\
-                \n  SRQ for Syntax error: {}\n  SRQ for Internal error: {}\
-                \n  SRQ Front Panel button: {}\
-                \n  SRQ for Cal err: {}\n  SQR for Power on: {}\
-                \nError information: \n  Calibration: {} \n  RAM: {}\n  ROM: {}\
-                \n  AD Slope: {}\n  AD: {}\n  AD-Link: {} \
-                \nDAC value: {}".format(
-                cur_mode, r_str, 6 - self.digits, self.int_trig, self.ext_trig,
-                self.auto_range, self.auto_zero, self.fifty_hz,
-                self.front_rear, self.cal_enable, self.SRQ_data_rdy,
-                self.SRQ_syntax_error, self.SRQ_internal_error,
-                self.SRQ_front_panel, self.SRQ_cal_error, self.SRQ_PON,
-                self.ERR_cal, self.ERR_RAM, self.ERR_ROM, self.ERR_slope,
-                self.ERR_AD, self.ERR_AD_Link, self.DAC_value)
-        )
-
-
-class Status(ctypes.Union):
-    """Union type element for the decoding of the status bit-fields
-    """
-    _fields_ = [
-        ("B", Status_bytes),
-        ("b", Status_bits)
-    ]
 
-
-class HP3478A(Instrument):
-    """ Represents the Hewlett Packard 3748A 5 1/2 digit multimeter
+class HP3478A(HPLegacyInstrument):
+    """ Represents the Hewlett Packard 3478A 5 1/2 digit multimeter
     and provides a high-level interface for interacting
     with the instrument.
     """
+    status_desc = Status
 
     def __init__(self, resourceName, **kwargs):
         kwargs.setdefault('read_termination', '\r\n')
         kwargs.setdefault('send_end', True)
         super().__init__(
             resourceName,
             "Hewlett-Packard HP3478A",
-            includeSCPI=False,
-            **kwargs
+            status_bitfield=self.status_desc,
```

Comment:
```suggestion
```
This line is not needed, since parameter `status_bitfield` has been replaced by class variable `status_desc`